submission/npc-comment-blocker/content.js

```diff
@@ -0,0 +1,203 @@
+const keywords = [

```

Comment:
This doesn't seem like a reliable way of identifying bot comments. You could have a [model trained on multiple youtube comments](https://stats.stackexchange.com/questions/49363/simple-machine-learning-bot-detection) or just use ChatGPT (see screenshot below)
<img width="982" alt="image" src="https://github.com/user-attachments/assets/101a60d5-a364-45f3-948b-787365d958f0">

Reply:
ok to both of these points 
making a chatgpt wrapper is just a bad thing to be fair and asking it for every comment would be really poor design 
for the training model , that could be an option however considering that i dont have enough data *yet* to do that 
the keyword regex matching design works effortlessly atleast for now and has a good accuracy score 
although there may be around 5% false positives aswell however its quite accurate 95% of the time and works well
i personally use it and a small group of people i met on reddit do aswell. Appreciate your suggestions however for now i believe this is the best solution while in the future ( i am in the process of collecting more spam across different languages ) a trained model could be an option.

Reply:
I've seen a few comments from the linked YouTube videos that seemed legit but were hidden by your extension. I'm not sure how you measured accuracy... can you provide a bit more details?

If you're concerned about usage. Perhaps collecting multiple comments at the same time and sending a single ChatGPT query could work? Or you could search for APIs that can do better classification that a simple regex matching. 

could you also link the Reddit post you did? Just genuinely interested in seeing what other people's opinions are.

Reply:
@JosiasAurel i further did some research on the stack post you recommended and it seems i am going to be doing that however running a entire model on client side is going to be resource intensive and cause issues for some people 
also i made sure to install this extension on all my school's pcs aswell and they will definately blow up if i have to run a model.
The other option being a server is not feasible to me due to pricing issues , please understand.
Although there is scope for using something like bayes classifier that approach will take slightly more time getting past the deadline of nov 30th